numpy_challenge/numpy_challenge.py

```diff
@@ -0,0 +1,132 @@
+import numpy as np
+
+
+def type_your_input_array(INPUT_ARRAY_TEXT):
+    return np.array([(row.split(",")) for row in input(INPUT_ARRAY_TEXT).split(";")], dtype=int)
+
+
+def make_my_favorite_arrays():
+    arrays_list = [np.zeros(9), np.arange(3, 28, 3).reshape(3, 3), np.eye(4)]
+    for i in range(3):
+        print(f'My favorite array {i + 1} is \n{arrays_list[i]}\n')
+
+
+def matrix_multiplication(matrix_1, matrix_2):
+    return np.matmul(matrix_1, matrix_2)
+
+
+def multiplication_check(array_list):
+    if multiply_matrices(array_list) is None:
+        return False
+    else:
+        return True
+
+
+def multiply_matrices(array_list):
+    product_matrix = array_list[0]
+    for i in range(1, len(array_list)):
+        columns_1 = product_matrix.shape[1]
+        rows_2 = array_list[i].shape[0]
+        if columns_1 == rows_2:
+            product_matrix = matrix_multiplication(product_matrix, array_list[i])
+        else:
+            return None
+    return product_matrix
+
+
+def compute_2d_distance(array_1, array_2):
+    if array_1.size == array_2.size == 2:
+        return np.sqrt(np.sum((array_1 - array_2) ** 2))
+    else:
+        return 'This arrays are not 2d!'

```

Comment:
Возвращать строку не лучшая идея. Лучше сделать вот так
```suggestion
        print('This arrays are not 2d!')
        return None

```